File: MATLAB/Matrix2Vec.m
function dFCstream_2D = Matrix2Vec(dFCstream_3D)

% FUNCTION dFCstream_2D = Matrix2Vec(dFCstream_3D)
% takes '3D' dFCstream as input and convert it to '2D' dFCstream
%
% Reference: Lucas Arbabyazd et al. (2020) MethodsX

if (ndims(dFCstream_3D) == 2)
    disp('You do not need this function since your dFCstream is alread 2D')
    return
end

n = size(dFCstream_3D, 1);
l = n*(n-1)/2;
F = size(dFCstream_3D, 3);
xo = find(tril(ones(n),-1));

dFCstream_2D = zeros(l, F);

for f = 1:F
    fc = dFCstream_3D(:,:,f);
    dFCstream_2D(:,f) = fc(xo);
end

File: Python/Matrix2Vec.py
#!/usr/bin/env python
# coding: utf-8

# In[1]:

# This function takes as input the '3D' dFCstream and convert it to '2D' dFCstream.


# Reference: Lucas Arbabyazd et al. (2020) MethodsX

import numpy as np

def Matrix2Vec(dFCstream_3D):
    
    if len(dFCstream_3D.shape)==2:
        return print('You do not need this function since your dFCstream is alread 2D')
    
    n = dFCstream_3D.shape[0]
    l = n*(n-1)/2
    F = dFCstream_3D.shape[2]
    xo = np.triu_indices(n, k = 1)
    
    dFCstream_2D = np.zeros((int(l), int(F)));
                  
    for f in range(F):
        fc = dFCstream_3D[:,:,f]
        dFCstream_2D[:,f] = fc[xo]
        
    return dFCstream_2D



